pattern/constraint/thing/ValueConstraint.java

```diff
@@ -181,6 +181,11 @@ public Long(ThingVariable owner, GraqlToken.Predicate.Equality predicate, long v
             super(owner, predicate, value, set());
         }
 
+        @Override
+        public GraqlToken.Predicate.Equality predicate() {
+            return super.predicate().asEquality();
+        }
+
```

Comment:
It's logical that a predicated from Double, Long, Boolean or DateTime Constraint should always return an equality predicate.